core/memory_tracker/README.md

```diff
@@ -0,0 +1,64 @@
+# Track Coherent Memory
+
+## Problem and Solution
+Coherent memories are implicitly synchronized which means there is no *flush*
+or *invalid* command to explicitly *write* or *read* to/from those memories.
+Henceforth we cannot rely on intercepting the *flush* and *invalid* commands to
+track the modifications for coherent memory.
+
+The solution is to utilize segmentation fault signal handler to tell to which
```

Comment:
This is only true on  Posix-based systems. We have to use other methods on other operating systems. I think it would make sense to describe this slightly differently, or at least a bit more explicitly.

Even something like:
The following explanation is for Posix-based systems. We intend to implement a similar system for Windows that uses SetUnhandledExceptionFilter.